libs/components/forms/src/lib/modules/radio/radio.component.ts

```diff
@@ -215,66 +232,78 @@ export class SkyRadioComponent implements OnDestroy, ControlValueAccessor {
   @Output()
   // eslint-disable-next-line @angular-eslint/no-output-native
   public get change(): Observable<SkyRadioChange> {
-    return this._change;
+    return this.#_change;
   }
 
   /**
    * Fires when the selected value changes.
    */
   @Output()
   public get checkedChange(): Observable<boolean> {
-    return this._checkedChange;
+    return this.#checkedChangeObs;
   }
 
   /**
    * Fires when the selected value changes.
    */
   @Output()
   public get disabledChange(): Observable<boolean> {
-    return this._disabledChange;
-  }
-
-  public get inputId(): string {
-    return `sky-radio-${this.id}-input`;
+    return this.#disabledChangeObs;
   }
 
   public set selectedValue(value: any) {
-    if (value !== this._selectedValue) {
-      this._selectedValue = value;
+    if (value !== this.#_selectedValue) {
+      this.#_selectedValue = value;
     }
   }
   public get selectedValue(): any {
-    return this._selectedValue;
+    return this.#_selectedValue;
   }
 
-  public radioGroupDisabled = false;
+  public inputId = '';
 
-  private _change = new EventEmitter<SkyRadioChange>();
-  private _checked = false;
-  private _checkedChange = new BehaviorSubject<boolean>(this._checked);
-  private _disabled = false;
-  private _disabledChange = new BehaviorSubject<boolean>(this._disabled);
-  private _name: string;
-  private _radioType: string;
-  private _selectedValue: any;
-  private _tabindex: number;
-  private _value: any;
+  public radioGroupDisabled = false;
 
-  constructor(private changeDetector: ChangeDetectorRef) {}
+  #checkedChange: BehaviorSubject<boolean>;
+  #checkedChangeObs: Observable<boolean>;
+  #defaultId = `sky-radio-${++nextUniqueId}`;
+  #disabledChange: BehaviorSubject<boolean>;
+  #disabledChangeObs: Observable<boolean>;
+
+  #_change = new EventEmitter<SkyRadioChange>();
+  #_checked = false;
+  #_disabled = false;
+  #_name: string | undefined;
+  #_radioType = 'info';
+  #_selectedValue: unknown;
+  #_tabindex = 0;
+  #_value: unknown;
+
+  #changeDetector: ChangeDetectorRef;
+
+  constructor(changeDetector: ChangeDetectorRef) {
+    this.#changeDetector = changeDetector;
+    this.#checkedChange = new BehaviorSubject<boolean>(this.checked);
+    this.#checkedChangeObs = this.#checkedChange.asObservable();
+    this.#disabledChange = new BehaviorSubject<boolean>(this.disabled);
+    this.#disabledChangeObs = this.#disabledChange.asObservable();
```

Comment:
☝️ These observables need to be completed in the `ngOnDestroy`.

Reply:
Fixed